tests/e2e/testutils.go

```diff
@@ -27,53 +29,56 @@ type Pod struct {
 	Node      string
 }
 
-//Runs command passed from within the node ServerIP
-func RunCmdOnNode(cmd string, nodename string) (string, error) {
-	runcmd := "vagrant ssh server-0 -c " + cmd
-	c := exec.Command("bash", "-c", runcmd)
-
-	var out bytes.Buffer
-	var errOut bytes.Buffer
-	c.Stdout = &out
-	c.Stderr = &errOut
-	if err := c.Run(); err != nil {
-		return errOut.String(), err
-	}
-	return out.String(), nil
-}
-
-// RunCommand Runs command on the cluster accessing the cluster through kubeconfig file
-func RunCommand(cmd string) (string, error) {
-	c := exec.Command("bash", "-c", cmd)
-	var out bytes.Buffer
-	c.Stdout = &out
-	if err := c.Run(); err != nil {
-		return "", err
+func CountOfStringInSlice(str string, pods []Pod) int {
+	count := 0
+	for _, pod := range pods {
+		if strings.Contains(pod.Name, str) {
+			count++
+		}
 	}
-	return out.String(), nil
+	return count
 }
 
-func CreateCluster(nodeos string, serverCount int, agentCount int) ([]string, []string, error) {
-	serverNodenames := make([]string, serverCount+1)
+func CreateCluster(nodeOS string, serverCount int, agentCount int, installType string) ([]string, []string, error) {
+	serverNodeNames := make([]string, serverCount+1)
 	for i := 0; i < serverCount; i++ {
-		serverNodenames[i] = "server-" + strconv.Itoa(i)
+		serverNodeNames[i] = "server-" + strconv.Itoa(i)
 	}
-
-	agentNodenames := make([]string, agentCount+1)
+	agentNodeNames := make([]string, agentCount+1)
 	for i := 0; i < agentCount; i++ {
-		agentNodenames[i] = "agent-" + strconv.Itoa(i)
+		agentNodeNames[i] = "agent-" + strconv.Itoa(i)
 	}
-	nodeRoles := strings.Join(serverNodenames, " ") + strings.Join(agentNodenames, " ")
+	nodeRoles := strings.Join(serverNodeNames, " ") + strings.Join(agentNodeNames, " ")
 	nodeRoles = strings.TrimSpace(nodeRoles)
-	nodeBoxes := strings.Repeat(nodeos+" ", serverCount+agentCount)
+	nodeBoxes := strings.Repeat(nodeOS+" ", serverCount+agentCount)
 	nodeBoxes = strings.TrimSpace(nodeBoxes)
-	cmd := fmt.Sprintf("NODE_ROLES=\"%s\" NODE_BOXES=\"%s\" vagrant up &> vagrant.log", nodeRoles, nodeBoxes)
-	if out, err := RunCommand(cmd); err != nil {
-		fmt.Println("Error Creating Cluster", out)
+	cmd := fmt.Sprintf("NODE_ROLES=\"%s\" NODE_BOXES=\"%s\" %s vagrant up &> vagrant.log", nodeRoles, nodeBoxes, installType)
+	fmt.Println(cmd)
+	if _, err := RunCommand(cmd); err != nil {
+		fmt.Println("Error Creating Cluster", err)
 		return nil, nil, err
 	}
+	return serverNodeNames, agentNodeNames, nil
+}
 
-	return serverNodenames, agentNodenames, nil
+func DeployWorkload(workload string, kubeconfig string, arch bool) (string, error) {
+	resource_dir := "../amd64_resource_files"
+	if arch {
+		resource_dir = "../arm64_resource_files"
+	}
+	files, err := ioutil.ReadDir(resource_dir)
+	if err != nil {
+		fmt.Println("Unable to read resource manifest file for ", workload)
```

Comment:
Should we return the error here since the call failed and files won't be useful?